### PR TITLE
Lidar and optical flow in sitl gazebo

### PR DESCRIPTION
**Pull request by ChristophTobler**
All changes for the sensors in sitl: models, plugins, etc.

### CODE REVIEW

File: CMakeLists.txt~ [hunk 1]
```diff
@@ -0,0 +1,75 @@
+cmake_minimum_required(VERSION 2.8.8 FATAL_ERROR)
```
Comment (LorenzMeier) [inline CMakeLists.txt~]: This is a temporary file and should not have been committed.

**Comment by mhkabir** [comment]
@LorenzMeier @ChristophTobler It'd be great if we could get this in soon. It would accelerate testing to no end!
Let me know if I can do anything either here, or on the Firmware side to help.

File: include/gazebo_opticalFlow_plugin.h~ [hunk 2]
```diff
@@ -0,0 +1,87 @@
+/*
```
Comment (mhkabir) [inline include/gazebo_opticalFlow_plugin.h~]: Temporary file, shouldn't have been committed. I expect you're using gedit, or some other text editor which creates these ```~```files. 
You should turn that off, or even better, use a different editor which doesn't use these.

File: models/flow_cam/model.sdf~ [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" ?>
```
Comment (mhkabir) [inline models/flow_cam/model.sdf~]: Temp file

File: models/iris_opt_flow/iris.sdf~ [hunk 4]
```diff
@@ -0,0 +1,475 @@
+<sdf version='1.5'>
```
Comment (mhkabir) [inline models/iris_opt_flow/iris.sdf~]: Temp file

File: models/iris_opt_flow/iris_opt_flow.sdf~ [hunk 5]
```diff
@@ -0,0 +1,442 @@
+<sdf version='1.5'>
```
Comment (mhkabir) [inline models/iris_opt_flow/iris_opt_flow.sdf~]: Temp file

File: models/iris_opt_flow/model.config~ [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
```
Comment (mhkabir) [inline models/iris_opt_flow/model.config~]: Temp file

File: models/lidar/model.sdf~ [hunk 7]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" ?>
```
Comment (mhkabir) [inline models/lidar/model.sdf~]: Temp file

File: msgs/CMakeLists.txt~ [hunk 8]
```diff
@@ -0,0 +1,33 @@
+find_package(Protobuf REQUIRED)
```
Comment (mhkabir) [inline msgs/CMakeLists.txt~]: Temp file

File: src/gazebo_mavlink_interface.cpp [hunk 9]
```diff
@@ -295,7 +301,7 @@ void GazeboMavlinkInterface::ImuCallback(ImuPtr& imu_message) {
 
   math::Pose T_W_I = model_->GetWorldPose();
   math::Vector3 pos_W_I = T_W_I.pos;  // Use the models'world position for GPS and pressure alt.
-
+
```
Comment (mhkabir) [inline src/gazebo_mavlink_interface.cpp]: Whitespace

File: worlds/iris_opt_flow.world~ [hunk 10]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" ?>
```
Comment (mhkabir) [inline worlds/iris_opt_flow.world~]: Temp file

**Comment by ChristophTobler** [comment]
hope this new commit does the job...

**Comment by mhkabir** [comment]
@ChristophTobler Looks good!

**Pull request merged**